ldapAdapterEXT-ext-impl/src/main/java/com/bosch/osmi/sw360/ldapAdapter/CustomPortalLDAPImporterImpl.java

```diff
@@ -131,5 +164,80 @@ private Organization addOrganization(String organizationName, User user) throws
 		}
         return organization;
 	}
+
+	private String mapOrganizationName(String name) {
+		loadLDAPImporterProperties();
+		String defaultTarget = defaultToNull ? null : name;
+
+		if (matchPrefix) {
+			return organizationMappings.keySet().stream()
+					.filter(name::startsWith)
+					.sorted(Comparator.comparing(String::length).reversed())
+					.findFirst()
+					.map(organizationMappings::get)
+					.orElse(defaultTarget);
+		} else { // match complete name
+			return organizationMappings.getOrDefault(name, defaultTarget);
+		}
+	}
+
+	private void loadLDAPImporterProperties() {
+		Properties ldapConfigProperties = loadProperties(CustomPortalLDAPImporterImpl.class, PROPERTIES_FILE_PATH);
+		matchPrefix = Boolean.parseBoolean(ldapConfigProperties.getProperty(MATCH_PREFIX_KEY, "false"));
+		defaultToNull = Boolean.parseBoolean(ldapConfigProperties.getProperty(DEFAULT_TO_NULL_KEY, "false"));
+
+		List<Object> mappingSourceKeys = ldapConfigProperties
+				.keySet()
+				.stream()
+				.filter(p -> ((String) p).startsWith(MAPPING_KEYS_PREFIX) && !((String) p).endsWith(MAPPING_VALUES_SUFFIX))
+				.collect(Collectors.toList());
+
+		organizationMappings = new HashMap<>();
+		for (Object sourceKey : mappingSourceKeys) {
+			String sourceOrg = ldapConfigProperties.getProperty((String)sourceKey);
+			String targetOrg = ldapConfigProperties.getProperty(sourceKey+MAPPING_VALUES_SUFFIX);
+			if (sourceOrg != null && targetOrg != null && sourceOrg.length() > 0 && targetOrg.length() > 0) {
+				organizationMappings.put(sourceOrg, targetOrg);
+			}
+		}
```

Comment:
Maybe already sort at this point and save a `List<Entry<,>>` instead of `HashMap<>`